packages/test/src/test-signals-are-always-delivered.ts

```diff
@@ -0,0 +1,48 @@
+import test from 'ava';
+import { WorkflowClient } from '@temporalio/client';
+import { Worker, DefaultLogger, Core, InjectedSinks } from '@temporalio/worker';
+import { defaultOptions } from './mock-native-worker';
+import { RUN_INTEGRATION_TESTS } from './helpers';
+import * as workflows from './workflows';
+
+if (RUN_INTEGRATION_TESTS) {
+  test.before(async () => {
+    await Core.install({ logger: new DefaultLogger('DEBUG') });
+  });
+
+  test('Signals are always delivered', async (t) => {
```

Comment:
Can you confirm this exact test as written fails without your fix? I am not seeing an assertion that signal is ever sent or received.

Should you at least return the counter from the workflow, or set an external variable from the sink call that proves it was run, or something similar?

Reply:
The workflow checks the counter is set.
I wrote the test before the fix.
What happens if the signal isn't received is that the workflow will never be able to complete because it'll keep receiving signals until the history exceeds the limit.
I'll clarify and maybe make this more explicit in the test.